Automattic/block-experiments
# Mobile: Add simple layout grid controls

This Pr is the complete collection of the controls we want to ship as the first iteration of the Grid Layout block on mobile.

Related Gutenberg Mobile PR: 
https://github.com/wordpress-mobile/gutenberg-mobile/pull/3444

iOS PR:
https://github.com/wordpress-mobile/WordPress-iOS/pull/16423

Android PR:
https://github.com/wordpress-mobile/WordPress-Android/pull/14578

## How has this been tested?
Test the full app via. 

1. Open the App 
2. Insert the Layout grid block. 
3. Change the number of columns via the block settings. 
4. Change the full-width alignment. 
5. Change the vertical alignment. Each of the individual columns should have had the vertical alignment updated.
6. Add content to columns. 
7. Change the padding of individual columns. Does the preview of the new padding look as expected? 
8. Change the vertical alignment of the content. Does it work as expected? 


To test on the web. 
Run 
```
yarn install
yarn build
yarn test

// remove the node_modules folder other wise the plugin is pretty big and might not install as expected
rm -rf node_modules
```

Create a zip of the block-experiments folder and install it as a plugin on your WordPress site. 
Activate block experiments plugin that you just install.

Does the layout grid block still work as expected for you?
Are you able to switch between different column types.
Layout Grid:
Are you able to change Width, change vertical alignment,  gutter sizes. different responsive columns?

Single column: 
Able to set padding, able to set background colours? 

On the web, you shouldn't notice any breaking changes.

## Screenshots 

https://user-images.githubusercontent.com/115071/118299226-568add00-b495-11eb-906c-5038ed142390.mp4
